File: cpp/svg/common/svgLine.cpp
// svgLine.cpp

#include "svgLine.h"

#include "xml.h"

svgLine::svgLine( float x1 , float y1 , 
                  float x2 , float y2 , 
                  string colorStroke , float widthStroke ,
                  float opacity ) 
: _x1( x1 )                
, _y1( y1 )              
, _x2( x2 )                  
, _y2( y2 )                  
, _colorStroke( colorStroke )        
, _widthStroke( widthStroke )    
, _opacity( opacity )    
{
}

svgLine::svgLine( Point& point1 , 
                  Point& point2 , 
                  string colorStroke , float widthStroke ,
                  float opacity ) 
: _x1( point1.xVal )                
, _y1( point1.yVal )              
, _x2( point2.xVal )                  
, _y2( point2.yVal )                  
, _colorStroke( colorStroke )        
, _widthStroke( widthStroke )    
, _opacity( opacity )    
{
}

void svgLine::setEndpoints( float x1 , float y1 , float x2 , float y2 )
{
   _x1 = x1 ;                
   _y1 = y1 ;              
   _x2 = x2 ;                  
   _y2 = y2 ;                  
}

void svgLine::draw( void ) 
{
  xmlStartMultitag( "line" ) ;
  xmlWritePar( "x1" , _x1 ) ;
  xmlWritePar( "x2" , _x2 ) ;
  xmlWritePar( "y1" , _y1 ) ;
  xmlWritePar( "y2" , _y2 ) ;

  xmlStartQuote( "style" ) ;

  xmlWriteColon( "stroke" , _colorStroke.c_str() ) ;
  xmlWriteColon( "stroke-width" , (int)_widthStroke ) ;
  xmlWriteColon( "stroke-opacity" , _opacity ) ;

  xmlEndQuote() ;

  xmlEndMultitag() ;
}

File: python/svg/svgLINE.py
import XML as xml
from svgBASE import svgBase

class svgLine(svgBase):

  def __init__( self , x1=0 , y1=0 , x2=100 , y2=100 , colorStroke="black" , widthStroke=1 , opacity=1.0 ):
    self._x1           = x1
    self._y1           = y1
    self._x2           = x2
    self._y2           = y2
    self._colorStroke  = colorStroke
    self._widthStroke  = widthStroke
    self._opacity      = opacity    

  def setEndpoints( self , x1 , y1 , x2 , y2 ):
    self._x1 = x1
    self._y1 = y1
    self._x2 = x2
    self._y2 = y2

  def draw( self ):
    xml.StartMultitag( "line" ) ;
    xml.WritePar( "x1" , self._x1 )
    xml.WritePar( "x2" , self._x2 )
    xml.WritePar( "y1" , self._y1 )
    xml.WritePar( "y2" , self._y2 )
    xml.StartQuote( "style" )
    xml.WriteColon( "stroke" , self._colorStroke )
    xml.WriteColon( "stroke-width" , self._widthStroke )
    xml.WriteColon( "stroke-opacity" , self._opacity ) 
    xml.EndQuote() ;
    xml.EndMultitag() ;

